reporter.js

```diff
@@ -16,7 +17,7 @@ var defaultReporter = function (errors) {
       return error.message
     }).join('\n\n')
 
-    gutil.log(allErrors)
+    fancyLog.log(allErrors)
```

Comment:
As we discussed yesterday, `fancy-log` does not have a `.log()` method [in its API](https://github.com/js-cli/fancy-log#api). Therefore, if we use the following task in `gulpfile.js`…

```js
gulp.task('test', function () {
  return gulp
    .src(['./**/*.pug'])
    .pipe(pugLint())
    .pipe(pugLint.reporter()) // this line causes `type == null` to evaluate to `true`
})
```

…this plugin fails with the following error:

```sh
$ ./node_modules/.bin/gulp
[14:28:25] Using gulpfile path/to/project/gulp-pug-linter/gulpfile.js
[14:28:25] Starting 'test'...
path/to/project/gulp-pug-linter/reporter.js:20
    fancyLog.log(allErrors)
             ^

TypeError: fancyLog.log is not a function
    at DestroyableTransform.defaultReporter (path/to/project/gulp-pug-linter/reporter.js:20:14)
    at DestroyableTransform._flush (path/to/project/gulp-pug-linter/reporter.js:90:14)
    at DestroyableTransform.<anonymous> (path/to/project/gulp-pug-linter/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:115:49)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:106:13)
    at DestroyableTransform.emit (events.js:208:7)
    at prefinish (path/to/project/gulp-pug-linter/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:493:12)
    at finishMaybe (path/to/project/gulp-pug-linter/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:501:7)
    at endWritable (path/to/project/gulp-pug-linter/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:513:3)
    at DestroyableTransform.Writable.end (path/to/project/gulp-pug-linter/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:483:41)
```

I am going to cover this usecase in a unit test. Otherwise, the 100% coverage is very deceptive.

Reply:
Fixed here: https://github.com/ilyakam/gulp-pug-linter/commit/3b9c59c039968ea0b8c7cdf48cffcb0859fbad96#diff-fdfe8a4d956efc7b79e8d0c19bf2e0f6R20